site/source/docs/porting/pthreads.rst

```diff
@@ -4,50 +4,58 @@
-  USE_PTHREADS=1`` and the other with ``-s USE_PTHREADS=0``.
-
-- When the linker flag `-s PTHREAD_POOL_SIZE=<integer>` is not specified and `pthread_create()` is called, the new thread will not actually start to run immediately, but the main JS thread must yield execution back to browser first. This behavior is a result of `#1049079 <https://bugzilla.mozilla.org/show_bug.cgi?id=1049079>`_.
+- When the linker flag ``-s PTHREAD_POOL_SIZE=<integer>`` is not specified and ``pthread_create()`` is called, the new thread will not actually start to run immediately, but you must yield execution back to the browser's main event loop first (in order for ``new Worker`` and the related events to happen). If you don't wait but synchronously expect it to be ready (like if you join on it, or if you look at memory where you expect it to write) things may not work. Using a pool allows things to work more like a native build.
 
-- Currently several of the functions in the C runtime, such as filesystem functions like `fopen()`, `fread()`, `printf()`, `fprintf()` etc. are not multithreaded, but instead their execution is proxied over to the main application thread. Memory allocation via `malloc()` and `free()` is fully multithreaded though. This proxying can generate a deadlock in a special situation that native code running pthreads does not have. See `bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for more information and how to work around this until proxying is no longer needed in Emscripten.
+- Currently several of the functions in the C runtime, such as filesystem functions like ``fopen()``, ``fread()``, ``printf()``, ``fprintf()`` etc. are not multithreaded, but instead their execution is proxied over to the main application thread. Memory allocation via ``malloc()`` and ``free()`` is fully multithreaded though. This proxying can generate a deadlock in a special situation that native code running pthreads does not have. See `bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for more information and how to work around this until proxying is no longer needed in Emscripten.
 
-- In order to keep proxying as responsive as possible, whenever main thread calls to a function that performs a futex wait, e.g. `usleep()`, `emscripten_futex_wait()`, or `pthread_mutex_lock()`, the wait is done in very short time slices, which means that functions such as `usleep()` are not necessarily effective to conserve power. In order to save battery in the main thread, it is best to yield back to the browser runtime. When a pthread perform a futex wait, it sleeps in considerably longer slices.
+- When doing a futex wait, e.g. ``usleep()``, ``emscripten_futex_wait()``, or ``pthread_mutex_lock()``, we use ``Atomics.wait`` on workers, which the browser should do pretty efficiently. But that is not available on the main thread, and so we busy-wait there. This is not recommended because it freezes the tab, and also wastes power.
```

Comment:
"This" seems a bit ambiguous to me. Maybe "Use of these functions" instead? Also I think power is still a red herring, because continuous spinning starts to affect the UI experience a long time before power?

Reply:
Editing to remove the ambiguity, thanks.

"freezes the tab" is already mentioned before "wastes power", is that not clear enough?

Reply:
Yeah true, that's probably fine as-is.